pkgs/applications/science/electronics/kicad/base.nix

```diff
@@ -185,7 +189,8 @@ stdenv.mkDerivation rec {
   HOME = "$TMP";
 
   # debug builds fail all but the python test
-  doInstallCheck = !(debug);
+  doCheck = (stable || testing);
+  doInstallCheck = !debug && (stable || testing);
```

Comment:
I don't understand what's going on here. Is this related to your comment about "The test failed due to lack of X server."?

Is the comment on line 191 ("debug builds fail all but the python test") still accurate? It feels like it has bit-rotted in light of the changes here.

Reply:
> I don't understand what's going on here. Is this related to your comment about "The test failed due to lack of X server."?

Yes

> Is the comment on line 191 ("debug builds fail all but the python test") still accurate? It feels like it has bit-rotted in light of the changes here.

No idea. The stable and testing build is broken due to libgit2 update so I cannot check it.

Reply:
I'll defer to @evils on this, but I believe we should make more of an effort to run the tests. Can we skip the problematic tests? If we cannot, we should leave a comment explaining why we're skipping the tests.

Reply:
I don't disagree but I lack knowledge how to debug tests in nix sandbox.

Reply:
Yeah, that's challenging. I find it can be helpful to get an interactive environment that's similar to the sandbox: https://wiki.nixos.org/wiki/Nixpkgs/Create_and_debug_packages#Using_nix-shell_for_package_development. That might help you?

Reply:
@jfly The problem is missing fontconfig file:

```
INFO     cli_util:utils.py:36 Executing command "kicad-cli sch export pdf --pages 1 -o /build/source/qa/tests/output/cli/basic_test.pone.pdf /build/source/qa/data/cli/basic_test/basic_test.kicad_sch"
_ test_sch_export_pythonbom[cli/basic_test/basic_test.kicad_sch-basic_test.pythonbom-6-cli_args0] _

kitest = <conftest.KiTestFixture object at 0x7fff6e6248c0>
test_file = 'cli/basic_test/basic_test.kicad_sch'
output_fn = 'basic_test.pythonbom', line_skip_count = 6, cli_args = []

    @pytest.mark.parametrize("test_file,output_fn,line_skip_count,cli_args",
                                [("cli/basic_test/basic_test.kicad_sch", "basic_test.pythonbom", 6, [])
                                 ])
    def test_sch_export_pythonbom( kitest,
                             test_file: str,
                             output_fn: str,
                             line_skip_count: int,
                             cli_args: List[str] ):
        input_file = kitest.get_data_file_path( test_file )
        compare_filepath = kitest.get_data_file_path( "cli/basic_test/{}".format( output_fn ) )
    
        output_filepath =  kitest.get_output_path( "cli/" ).joinpath( output_fn )
    
        command = ["kicad-cli", "sch", "export", "python-bom"]
        command.extend( cli_args )
        command.append( "-o" )
        command.append( str( output_filepath ) )
        command.append( input_file )
    
        stdout, stderr, exitcode = utils.run_and_capture( command )
    
        assert exitcode == 0
>       assert stderr == ''
E       AssertionError: assert 'Fontconfig e...ile: (null)\n' == ''
E         
E         + Fontconfig error: Cannot load default config file: No such file: (null)
```

Test passes in this environment

Reply:
How do I do debug build?

Reply:
Also any idea how do I turn off those tests. It takes me over an hour for trimmed build (just -unstable) so I move slowly and can look at this just 1-2 times/week.

Reply:
@jfly I disabled only the tests that fail and added note for debug build.